src/test/fuzz/block_index.cpp

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2023 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include <chain.h>
+#include <chainparams.h>
+#include <node/blockstorage.h>
+#include <txdb.h>
+#include <validation.h>
+#include <test/fuzz/FuzzedDataProvider.h>
+#include <test/fuzz/fuzz.h>
+#include <test/fuzz/util.h>
+#include <test/util/setup_common.h>
+
+namespace {
+
+const BasicTestingSetup* g_setup;
+
+// Hardcoded block hash and nBits to make sure the blocks we store pass the pow check.
+const uint256 g_block_hash{uint256S("000000002c05cc2e78923c34df87fd108b22221ac6076c18f3ade378a4d915e9")};
+uint32_t g_bits{0x1d00ffff};
+
+bool operator==(const CBlockFileInfo& a, const CBlockFileInfo& b)
+{
+    return a.nBlocks == b.nBlocks &&
+        a.nSize == b.nSize &&
+        a.nUndoSize == b.nUndoSize &&
+        a.nHeightFirst == b.nHeightFirst &&
+        a.nHeightLast == b.nHeightLast &&
+        a.nTimeFirst == b.nTimeFirst &&
+        a.nTimeLast == b.nTimeLast;
+}
+
+CBlockHeader ConsumeBlockHeader(FuzzedDataProvider& provider)
+{
+    CBlockHeader header;
+    header.nVersion = provider.ConsumeIntegral<decltype(header.nVersion)>();
+    header.hashPrevBlock = g_block_hash;
+    header.hashMerkleRoot = g_block_hash;
+    header.nTime = provider.ConsumeIntegral<decltype(header.nTime)>();
+    header.nBits = g_bits;
+    header.nNonce = provider.ConsumeIntegral<decltype(header.nNonce)>();
+    return header;
+}
+
+} // namespace
+
+void init_block_index()
+{
+    static const auto testing_setup = MakeNoLogFileContext<>(ChainType::MAIN);
+    g_setup = testing_setup.get();
+}
+
+FUZZ_TARGET(block_index, .init = init_block_index)
+{
+    FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
+    auto block_index = kernel::BlockTreeDB(DBParams{
```

Comment:
nit (feel free to ignore):

```diff
diff --git a/src/test/fuzz/block_index.cpp b/src/test/fuzz/block_index.cpp
index 7885eda747..3f2a55f310 100644
--- a/src/test/fuzz/block_index.cpp
+++ b/src/test/fuzz/block_index.cpp
@@ -54,11 +54,6 @@ void init_block_index()
 FUZZ_TARGET(block_index, .init = init_block_index)
 {
     FuzzedDataProvider fuzzed_data_provider{buffer.data(), buffer.size()};
-    auto block_index = kernel::BlockTreeDB(DBParams{
-        .path = "", // Memory only.
-        .cache_bytes = 1 << 20, // 1MB.
-        .memory_only = true,
-    });
 
     // Generate a number of block files to be stored in the index.
     int files_count = fuzzed_data_provider.ConsumeIntegralInRange(1, 100);
@@ -75,6 +70,12 @@ FUZZ_TARGET(block_index, .init = init_block_index)
         }
     }
 
+    auto block_index = kernel::BlockTreeDB(DBParams{
+        .path = "", // Memory only.
+        .cache_bytes = 1 << 20, // 1MB.
+        .memory_only = true,
+    });
+
     // Generate a number of block headers to be stored in the index.
     int blocks_count = fuzzed_data_provider.ConsumeIntegralInRange(files_count * 10, files_count * 100);
     std::vector<std::unique_ptr<CBlockIndex>> blocks;
```

we could initialize the DB after having the files.

Reply:
Heh, no that's a good point. There is a bunch happening in CDBWrapper's constructor so i'm sure this would reduce the time we spend on uninteresting inputs.

Reply:
Ok, looks like i'm wrong. It doesn't make any difference in runtime when running my target over my local corpus. I'll hold off making this change then.